sycl/doc/extensions/proposed/sycl_ext_oneapi_graph_fusion.asciidoc

```diff
@@ -0,0 +1,644 @@
+requires access to the data that would be stored into the internalized memory if
+no internalization were to happen.
+
+In sum this allows users to trigger internalization of a buffer or allocated
+device memory by just specifying a single property.
+
+==== Buffer internalization
+
+In some cases, the user will specify different internalization targets for a
+buffer and accessors to such buffer. When incompatible combinations are used, an
+`exception` with `errc::invalid` error code is thrown. Otherwise, these
+properties must be combined as follows:
+
+[options="header"]
+|===
+|Accessor Internalization Target|Buffer Internalization Target|Resulting Internalization Target
+
+.3+.^|None
+|None
+|None
+
+|Local
+|Local
+
+|Private
+|Private
+
+.3+.^|Local
+|None
+|Local
+
+|Local
+|Local
+
+|Private
+|*Error*
+
+.3+.^|Private
+|None
+|Private
+
+|Local
+|*Error*
+
+|Private
+|Private
+|===
+
+In case different internalization targets are used for accessors to the same
+buffer, the following (commutative and associative) rules are followed:
+
+[options="header"]
+|===
+|Accessor~1~ Internalization Target|Accessor~2~ Internalization Target|Resulting Internalization Target
+
+|None
+|_Any_
+|None
+
+.2+.^|Local
+|Local
+|Local
+
+|Private
+|None
+
+|Private
+|Private
+|Private
+|===
+
+If no work-group size is specified or two accessors specify different
+work-group sizes when using local internalization for any of the
+kernels involved in the fusion, no internalization will be
+performed. If there is a mismatch between the two accessors (access
+range, access offset, number of dimensions, data type), no
+internalization is performed.
+
+== Examples
+
+=== Buffer-based example
+
+```c++
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+
+struct AddKernel {
+  accessor<int, 1> accIn1;
+  accessor<int, 1> accIn2;
+  accessor<int, 1> accOut;
+
+  void operator()(id<1> i) const { accOut[i] = accIn1[i] + accIn2[i]; }
+};
+
+int main() {
+  constexpr size_t dataSize = 512;
+  int in1[dataSize], in2[dataSize], in3[dataSize], out[dataSize];
+
+  queue q{default_selector_v};
+
+  {
+    buffer<int> bIn1{in1, range{dataSize}};
+    bIn1.set_write_back(false);
+    buffer<int> bIn2{in2, range{dataSize}};
+    bIn2.set_write_back(false);
+    buffer<int> bIn3{in3, range{dataSize}};
+    bIn3.set_write_back(false);
+    buffer<int> bTmp1{range{dataSize}};
+    // Internalization specified on the buffer
+    buffer<int> bTmp2{
+        range{dataSize},
+        {sycl::ext::oneapi::experimental::property::promote_private{}}};
+    // Internalization specified on the buffer
+    buffer<int> bTmp3{
+        range{dataSize},
+        {sycl::ext::oneapi::experimental::property::promote_private{}}};
+    buffer<int> bOut{out, range{dataSize}};
+    bOut.set_write_back(false);
+
+    ext::oneapi::experimental::command_graph graph{
+        q.get_context(), q.get_device(),
+        sycl::ext::oneapi::experimental::property::graph::no_host_copy{}};
```

Comment:
This property no longer exist, should use `assume_buffer_outlives_graph` property instead

Reply:
Do we still need to call `set_writeback(false)` with that property applied to the graph or is that not necessary?

Reply:
Yes, `set_write_back` is still required even with `assume_buffer_outlives_graph`. Although we could relax this constraint, because the `assume_buffer_outlives_graph ` property will go away in the future once we implement extending buffer lifetimes, we don't want users to write code that depends on buffer copy-back, and then have to get them to rewrite that to be compatible with a later spec revision.